mkosi/__init__.py

```diff
@@ -2218,21 +2218,61 @@ def finalize_initrds(context: Context) -> list[Path]:
     return [build_default_initrd(context)]
 
 
+def expand_kernel_specifiers(text: str, kver: str, token: str, roothash: str, boot_count: str) -> str:
+    specifiers = {
+        "&": "&",
+        "e": token,
+        "k": kver,
+        "h": roothash,
+        "c": boot_count
+    }
+
+    def replacer(match: re.Match[str]) -> str:
+        m = match.group("specifier")
+        if specifier := specifiers.get(m):
+            return specifier
+
+        logging.warning(f"Unknown specifier '&{m}' found in {text}, ignoring")
+        return ""
+
+    return re.sub(r"&(?P<specifier>[&a-zA-Z])", replacer, text)
+
+
 def install_type1(
     context: Context,
     kver: str,
     kimg: Path,
     token: str,
     partitions: Sequence[Partition],
 ) -> None:
-    dst = context.root / "boot" / token / kver
-    entry = context.root / f"boot/loader/entries/{token}-{kver}.conf"
+    roothash = ""
+    if roothash_cli := finalize_roothash(partitions):
```

Comment:
since that specific value is the entire command line option for specifying the roothash and not just the value.

I guess "command line interface" is also not really right in that case.

Reply:
If it makes more sense I can keep that `roothash` and change the other one to `roothash_value`?

Reply:
Names are hard. :) I think I'd go for a simple `r`, since it's only used in this one place as a temporary variable. I have no strong preference, but the `_cli` did make me wonder.

Reply:
It is also used again https://github.com/NekkoDroid/mkosi/blob/1079fe42f40d818f818fadf340e1851abc3abfc3/mkosi/__init__.py#L2279 in the Type 1 case, so I think I'll go with `roothash_value` and `roothash`